docs/content/guides/developer/app-examples/trustless-swap.mdx

```diff
@@ -1,27 +1,1957 @@
+
+The first test works by creating an object to test, locking it and unlocking it – this should finish executing without aborting.
+The last two lines exist to keep the Move compiler happy by cleaning up the test coin and test scenario objects, because values in Move will not be implicitly cleaned up unless they have the `drop` ability.
+
+{@inject: examples/trading/contracts/escrow/sources/lock.move#fun=test_lock_unlock noComments}
+
+The other test is testing a failure scenario – that an abort happens. It creates two locked objects (this time the values are just `u64`s), and use the key from one to try and unlock the other, which should fail (specified using the `expected_failure` attribute).
+
+Unlike the previous test, the same clean-up is not needed, because the code is expected to abort. Instead, add another abort after the code that you expect to abort (making sure to use a different code for this second abort).
+
+{@inject: examples/trading/contracts/escrow/sources/lock.move#fun=test_lock_key_mismatch noComments}
+
+:::tip Additional resources
+
+- [Full source code](https://github.com/MystenLabs/sui/blob/705ee1ed3ce8cfadc4597c6facb6769d7dfb5896/examples/trading/contracts/escrow/sources/lock.move)
+- Concepts: [Test Scenario](../first-app/build-test.mdx#testing-a-package)
+
+:::
+
+### The Escrow protocol {#escrow}
+
+Trading proceeds in three steps:
+
+1. The first party locks the object they want to trade – this is already handled by the `lock` module you wrote earlier.
+1. The second party puts their object up for escrow and registers their interest in the first party's object. This will be handled by a new module – `escrow`.
+1. The first party completes the trade by providing their locked object and the key to unlock it. Assuming all checks pass, this transfers their object to the second party and makes the second party's object available to them.
+
+You can start by implementing steps two and three, by defining a new type to hold the escrowed object. It holds the `escrowed` object and an `id: UID` (because it's an object in its own right), but it also records the `sender` and intended `recipient` (to confirm they match when the trade happens), and it registers interest in the first party's object by recording the `ID` of the key that unlocks the `Locked<U>` that contains the object.
+
+{@inject: examples/trading/contracts/escrow/sources/shared.move#struct=Escrow noComments singleSpace}
+
+You also need to create a function for creating the `Escrow` object. The object is shared because it needs to be accessed by the address that created it (in case the object needs to be returned) and by the intended recipient (to complete the swap).
+
+<details>
+<summary>
+`create` function in `shared.move`
+</summary>
+{@inject: examples/trading/contracts/escrow/sources/shared.move#fun=create noComments}
```

Comment:
Would it be possible to not include the `event::emit` calls in the snippet at this point? Because we only explain them at a later point.